draft-gpew-priv-ppm.md

```diff
@@ -672,247 +678,381 @@ independently.
+
+The helper's response to the leader is an HTTP 200 OK whose body is the
+AggregateInitResp and media type is "message/ppm-aggregate-init-resp".
+
+Upon receipt of a helper's AggregateInitResp message, the leader checks that the
+sequence of PrepareStep messages corresponds to the ReportShare sequence of the
+AggregateInitReq. If any message appears out of order, is missing, has an
+unrecognized nonce, or if two messages have the same nonce, then the leader MUST
+abort with error "unrecognizedMessage".
+
+[[OPEN ISSUE: the leader behavior here is sort of bizarre -- to whom does it abort?]]
+
+#### Input Share Decryption {#input-share-decryption}
+
+Each report share has a corresponding task ID, nonce, list of extensions, and encrypted
+input share. Let `task_id`, `nonce`, `extensions`, and `encrypted_input_share` denote these
+values, respectively. Given these values, an aggregator decrypts the input
+share as follows. First, the aggregator looks up the HPKE config and corresponding
+secret key indicated by `encrypted_input_share.config_id`. If not found, then it
+marks the report share as invalid with the error `hpke-unknown-config-id`.
+Otherwise, it decrypts the payload with the following procedure:
 
 ~~~
-context = SetupBaseR(helper_share.enc, sk, task_id ||
+context = SetupBaseR(encrypted_input_share.enc, sk, task_id ||
                      "ppm-00 input share" || 0x01 || server_role)
-input_share = context.Open(nonce || extensions, helper_share)
+
+input_share = context.Open(nonce || extensions,
+                           encrypted_input_share.payload)
 ~~~
 
-where `sk` is the HPKE secret key, `task_id` is `AggregateReq.task_id` and
-`server_role` is the role of the server (`0x02` for the leader and `0x03` for
-the helper; `0x01`, the client role, is used as the sender). `nonce` and
-`extensions` are obtained from the corresponding fields in `AggregateSubReq`. If
-decryption fails, then the sub-response consists of a "decryption error" alert.
-[See issue#57.] Otherwise, the helper handles the request for its plaintext
-input share `input_share` and updates its state as specified by the PPM
-protocol.
+where `sk` is the HPKE secret key, `task_id` is the task ID, `nonce` and
+`extensions` are the nonce and extensions of the report share respectively,
+and `server_role` is 0x02 if the aggregator is the leader and 0x03 otherwise.
+If decryption fails, the aggregator marks the report share as invalid with the
+error `hpke-decrypt-error`. Otherwise, it outputs the resulting `input_share`.
 
-After processing all of the sub-requests, the helper encrypts its updated state
-and constructs its response to the aggregate request.
+#### Input Share Validation {#input-share-batch-validation}
 
-#### Leader State
+Validating an input share will either succeed or fail. In the case of failure,
+the input share is marked as invalid with a ReportShareError of type `vdaf-prep-error`.
```

Comment:
I think "vdaf-prep-error" here is incorrect. The two checks below each specify different ReportShareError codes. The next section, "Input Share Preparation", correctly says it may produce a "vdaf-prep-error" error.

Reply:
Yep, resolved. Thanks!